### PR TITLE
Rework documentation

### PR DESCRIPTION
**Pull request by charlydelta**
Screenshot from 2019-08-02 19-36-28

**With this PR I propose multiple changes to the docs, a lot of tidying up, and the start to formally define the architecture of this project to ease documenting the individual modules.**

Change overview:

- New Gothic-styled Sphinx theme
- Tidied up `rst` source files (exhausting line-length, whitespacing, ...)
- Efforts to unify capitalisation, fixed typos, `md` to `rst` conversion errors, etc.
- Integrated documentation for the new `core` module generated by Doxygen, which makes use of the grouping feature
- Documentation for the executables (only one there yet)
- Added cross references where ever applicable to ease navigation and potential relevant documentations in other chapters
- New category-based menu. The idea is to have a setup section (*Getting Started*), a tutorial section for devs (*Tutorials*), a section where insights of reverse engineering efforts are documented (*Case Studies*, better name maybe?), and a section where you can find concrete code reference (*Code Reference*). The code reference chapter should eventually include a landing page, where the architecture is illustrated very superficially. All consecutive subsections should showcase each individual module of the project. Last but not least, the executables are documented. At this stage of this effort, only one module is in the documentation and the start to document the executbales.
- Grouped `rst` source files according to the menu categories
- Old doc build artefacts were purged, and a complete new and clean build was added to `docs`
- Re-added doc on how to deal with the documentation itself. It takes some effort to get this up and running, and everyone who wants to contribute to the documentation should have as little of hassle as possible

What needs to be done (not in this PR though):
- **Formally define an architecture** and add the then defined modules to the *Code Reference* section
- More documentation on executables
- Maybe rework the *Case Studies* chapter and think about a better way to structure it

### CODE REVIEW

**Comment by ataulien** [comment]
I got this warning/error when compiling the docs:

```
WARNING: error while running plantuml

Current path is /home/andre/code/REGoth-bs/docs-source/case-studies/.
Cannot include style.iuml
Preprocessor Error: Cannot include style.iuml
ERROR
1
Cannot include style.iuml
Some diagram description contains errors
```

Should probably use a relative path here.

File: docs-source/getting-started/building-regoth.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  $ cd build
+
+  # Run CMake.
+  $ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
+
+Now we can build the project (if you want to use less cores than available, replace ``$(nproc)``
+with the desired number.  You might want to do this if you have little RAM, expect up to 2GB of RAM
+usage per core):
+
+.. code:: sh
```
Comment (ataulien) [inline docs-source/getting-started/building-regoth.rst]: Missing syntax highlighting. Should probably use `.. code-block:: sh` instead of `.. code:: sh`.

Comment (charlydelta) [reply docs-source/getting-started/building-regoth.rst]: My bad, of course!

File: docs-source/getting-started/building-docs.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
+Building the documentation
+==========================
+
+This documentation is written as reStructuredText and built using
```
Comment (ataulien) [inline docs-source/getting-started/building-docs.rst]: We should probably make clear here that this step can be skipped if you just want to get started.

Comment (charlydelta) [reply docs-source/getting-started/building-docs.rst]: Good point, done

File: docs-source/getting-started/running.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  ``REGoth-bs/lib/bsf/Dependencies/PhysX/lib/`` and copy all ``.so``-files to the location where
+  ``libbsf.so`` was built, usually ``REGoth/build/lib``.
+
+
+The asset cache
+---------------
+
+When loading a world for the first time, REGoth will convert the resources of the original Gothic
+game into a more modern format faster to load next time.  If you want to re-import everything,
+simply delete the `cache/` directory next to your REGoth executables.
```
Comment (ataulien) [inline docs-source/getting-started/running.rst]: Needs double backticks!

File: docs-source/tutorials/setting-up-app.rst [hunk 4]
```diff
@@ -1,35 +1,45 @@
...
 
-All of those Test-Applications reside directly inside the ``src``-directory, with a name of
-``main_something.cpp``.
+All of those test applications reside directly inside the ``src`` directory, with a name of
+``main_<Subject>.cpp``.  The existing test applications shipped with REGoth are documented
+:ref:`here <execs>`.
 
+.. note::
+
+  For an overview on the existing executables and what they do exactly, see
```
Comment (ataulien) [inline docs-source/tutorials/setting-up-app.rst]: That information is already provided in the paragraph above, so the note is redundant.

**Pull request merged**